strings/is_substring_palindrome.py

```diff
@@ -0,0 +1,61 @@
+MAX_LEN = 1000
+
+text = ""
+dp = [[False for i in range(MAX_LEN)] for i in range(MAX_LEN)]
+
+def preprocess(s):
+    """
+        Preprocesses a string using dynamic programming.
+        Time complexity: O(n^2)
+    """
+    global text
+    global dp
+    text = s
+    n = len(s)
+    dp = [[False for i in range(n)] for i in range(n)]
+
+    for i in range(n):
+        dp[i][i] = True
+
+    for i in range(n - 1):
+        dp[i][i + 1] = (text[i] == text[i + 1]) 
+
+    for substr_len in range(2, n + 1):
+        for i in range(n - substr_len + 1):
+            j = i + substr_len - 1
+            dp[i][j] = (text[i] == text[j] and dp[i + 1][j - 1])
+
+
+def is_substring_palindrome(l, r):
+    """

```

Comment:
Strange format of the docstring

Reply:
I updated the code. It's now OK. Please review it again.